tests/test_webhook.py

```diff
@@ -106,11 +106,15 @@ def test_webhook_handles_post_request_invalid_signature(
     ) in caplog.record_tuples
 
 
-def test_webhook_handles_post_request_not_job_end(caplog):
+def test_webhook_handles_post_request_not_job_end(caplog, mocked_valid_signature):
+    request_body = {
+        "action": "THIS_IS_WRONG",
+        "job_instance": {"name": "PPOD Export"},
+    }
```

Comment:
+1 to defining the request as a dict, than dumping to JSON later on.  Much easier to work with.